WordPress/Classes/Services/PostHelper.m

```diff
@@ -248,8 +266,11 @@ + (NSArray *)mergePosts:(NSArray <RemotePost *> *)remotePosts
                 post = [blog createPost];
             }
         }
-        [PostHelper updatePost:post withRemotePost:remotePost inContext:context];
-        [posts addObject:post];
+        // Only update if the post is original; don't update a revision with the remote post.
+        if (post.isOriginal) {
```

Comment:
(Nit) `[blog lookupPostWithForeignID:foreignID inContext:context];` should return an original since this whole method is expected to work only with originals.

Reply:
If the `foreignID` assignment is moved to post created (as suggested [here](https://github.com/wordpress-mobile/WordPress-iOS/pull/23173/files#r1595335402)), the `lookup(...)` method will start working with `original = NULL`, and will start returning originals.

Reply:
Fixed! [ea45f5a](https://github.com/wordpress-mobile/WordPress-iOS/pull/23173/commits/ea45f5a35a9e593285c803076e543107f5aed27b)